lib/logstash/inputs/jms.rb

```diff
@@ -300,6 +336,41 @@ def queue_event(msg, output_queue)
     end
   end
 
+  def to_string_keyed_hash(hash)
+    hash.inject({}) { |h, (key, val)| h[key.to_s] = val; h }
+  end
+  private :to_string_keyed_hash
+
+  # @param msg [JMS::MapMessage]
+  # @return [LogStash::Event]
+  def process_map_message(msg)
+    data = to_string_keyed_hash(msg.data)
+    do_target_check_once_and_get_event_factory.new_event(data)
+  end
+
+  def do_target_check_once_and_get_event_factory
+    @target_checked ||= begin
+                          do_target_check
+                          targeted_event_factory
+                        end
+  end
+
+  TARGET_NOT_SET_MESSAGE = LogStash::PluginMixins::ECSCompatibilitySupport::TargetCheck::TARGET_NOT_SET_MESSAGE
+
+  def do_target_check
+    return true unless target.nil?
+    return nil if ecs_compatibility == :disabled
+    logger.info(TARGET_NOT_SET_MESSAGE) # target isn't set in ECS mode
+  end
+
+  # @param msg [JMS::TextMessage, JMS::BytesMessage]
+  # @return [LogStash::Event, nil]
+  def decode_message(msg)
+    text = msg.to_s # javax.jms.TextMessage#getText (e.g. JSON payload)
+    event = nil
+    @codec.decode(text) { |e| event = e } unless text.nil?
```

Comment:
spidey senses: if our codec emits multiple, we use only the last. This is the same behaviour as previous but perhaps worth noting.

Reply:
```suggestion
    # if codec emits multiple, we use only the last event
    @codec.decode(text) { |e| event = e } unless text.nil?
```